pkg/reconciler/reconciler.go

```diff
@@ -234,6 +234,13 @@ func (r *Reconciler) updatePipelineRunToInProgress(ctx context.Context, logger *
 	if event.InstallationID > 0 {
 		event.Provider.WebhookSecret, _ = pac.GetCurrentNSWebhookSecret(ctx, r.kinteract, r.run)
 	} else {
+		// in ReconcileKind this r.secretNS is not set and it is needed when git provider
+		// is not github.
+		r.secretNS = repo.GetNamespace()
```

Comment:
Do we need to set this in reconciler field? can't we just pass the value while getting the secret, instead of setting it on reconciler?

Reply:
@piyush-garg I think `r.secretNS` is the field which is meant for fetching secret for PipelineRun, if you see its uses it is used only two times once in [reportFinalStatus](https://github.com/openshift-pipelines/pipelines-as-code/blob/2d465fe57657b6ad04e409fa8c951f29c2e33c77/pkg/reconciler/reconciler.go#L169) and in [updatePipelineRunToInProgress](https://github.com/openshift-pipelines/pipelines-as-code/blob/2d465fe57657b6ad04e409fa8c951f29c2e33c77/pkg/reconciler/reconciler.go#L244).

Reply:
but in case of global repo and all, setting this param can may be have adverse affect on next event etc, i would rather skip it for now like setting in reconciler

Reply:
I m not sure about the effect now but when i went through the code i see even r.secretNS is set either by `repo.GetNamespace()` or `r.secretNS = r.globalRepo.GetNamespace()` for globalRepo

@piyush-garg do you think it cause some issue :thinking:

Reply:
I see the point of @piyush-garg even if i don't think it would cause problem but i think @zakisk you can remove that variable being on the struct and instead be scoped to the function that would be at least a saner code.